vm/devices/tpm/src/resolver.rs

```diff
@@ -86,13 +73,11 @@ impl AsyncResolveResource<ChipsetDeviceHandleKind, TpmDeviceHandle> for TpmDevic
             None
         };
 
-        let ak_cert_type = match (get_attestation_report, request_ak_cert) {
-            (Some(get_attestation_report), Some(request_ak_cert)) => {
-                TpmAkCertType::HwAttested(get_attestation_report, request_ak_cert)
-            }
-            (None, Some(request_ak_cert)) => TpmAkCertType::Trusted(request_ak_cert),
-            (Some(_), None) => Err(ResolveTpmError::InvalidAkCertType)?,
-            (None, None) => TpmAkCertType::None,
+        let ak_cert_type = match (resource.support_attestation_report, request_ak_cert) {
+            (true, Some(request_ak_cert)) => TpmAkCertType::HwAttested(request_ak_cert),
+            (false, Some(request_ak_cert)) => TpmAkCertType::Trusted(request_ak_cert),
+            (true, None) => Err(ResolveTpmError::InvalidAkCertType)?,
```

Comment:
i.e: it should be possible to fully eliminate this statement by fusing those two `resource` fields into a single tre-state enum

Reply:
then we map enum (`TpmAkCertTypeResource`) to enum (`TpmAkCertType`) instead?

Reply:
right, I think so